### DISCUSSION TITLE
Remove data package in vault

### DISCUSSION THREAD

**Discussion by pradeep10kumar** [UtrechtUniversity/yoda#336] [Q&A] [answered]
Hi,

Can we remove a data package added to vault? If yes, how can it be done once it is added to vault?

**Comment by stsnel** [comment, marked as answer]
Hi Pradeep,

Technical admins (`rodsadmin` users) can remove a data package in the vault using the iCommands (e.g. `irm -r "/zoneName/home/vault-foo/..."`). As of Yoda 1.9.0, this operation can not (yet) be performed using the Yoda portal.

Best,
Sietse

**Comment by Danny-dK** [comment]
Note that in SURF context (after much trial and error) we have needed to first own the vault package if not our own package (even as rodsadmin). Generally our workflow is (again after much trial and error along with SURF admins) :

```
ichmod -r -V -M own john.doe@wur.nl "/zonename/home/vault-testdannyrmvault"

irm -r -V "/zonename/home/vault-testdannyrmvault/research-name[1676560437]"

irm -r -f -V "/zonename/home/vault-testdannyrmvault"
```
Note that john.doe@wur.nl should be your own admin email address. In case you are not able to delete the parent vault directory (vault-testdannyrmvault), then first do:

`ichmod -M write john.doe@wur.nl "/zonename/home/"`

**Do not use the `-r` flag in `ichmod -M write john.doe@wur.nl "/zonename/home/"`.** Then use previous commands to delete the vault parent directory (ichmod own the vault parent directory vault-testdannyrmvault and then irm the vault parent directory vault-testdannyrmvault). Once vault entry removed, do:

`ichmod -M null john.doe@wur.nl "/zonename/home/"`

**Comment by pradeep10kumar** [comment]
1. However, the example displayed includes the use of the "-r" flag in ichmod (cotrary to Do not use the -r flag in ichmod -M write john.doe@wur.nl "/zonename/home/"). Could you provide clarification?
2. Is the following command: ichmod -r -V -M own @.*** "/zonename/home/vault-testdannyrmvault" the procedure for owning the vault package?
3. Additionally, is it possible to delete specific files within the given folder rather than the entire directory?

**Comment by Danny-dK** [comment]
@pradeep10kumar 
1
The -r flag in:
```ichmod -r -V -M own john.doe@wur.nl "/zonename/home/vault-testdannyrmvault"```
Means recursive only on that specific vault entry and is therefore changing the ownership in a limited set of folders.

If you would use the -r flag in:
```ichmod -M write john.doe@wur.nl "/zonename/home/"```
You would recurse throughout all available folders within your Yoda instance research area and vault area, even for those where you are not focusing on or have any intend for changing ownership. 

2
Yes, you're adding your own email address as the owner for the vault package so that you are allowed to do certain things with that package as an admin (such as removal).

3
I'm not sure you can delete a specific file within a vault entry. @lwesterhof will be able to answer. Even if possible, I don't think you should delete a file from a vault submission. The intent of a vault submission is to preserve what was submitted at that point in time of submission. If you start removing files, the integrity of that submission is difficult to ascertain. Better to just re-submit data to the vault again and remove an obsolete vault submission in its entirety (eg the entire vault folder entry) if required.

**Reply by pradeep10kumar** [reply]
Thanks for your reply. I guess it is better to use :  ichmod -M write john.doe@wur.nl "/zone name/home/vault-testdannyrmvault" rather than ichmod -M write john.doe@wur.nl "/zonename/home/". It will be only limited to the targated folder.

**Reply by Danny-dK** [reply]
Not necessarily. As indicated, it may be a requirement when using a SURF hosted Yoda instance to set write privileges on 'home' (without recursing) before being able to delete a vault entry. In general, first try to do it only on the vault entry (follow the workflow only on the vault entry), if it fails to delete, you likely need to first set 'write' privileges on the home zone (but without recursing).
Note that in your first line you're not recursing, which is what you actually want to do there as the parent vault entry likely contain sub entries.

**Comment by pradeep10kumar** [comment]
So, ichmod -M write john.doe@wur.nl "/zonename/home/" with flag -r will recurse on all vault and research area. But without flag -r would recurse only in home zone. I am not sure about it as all my data folders (starting with research, vault, etc) are in the home zone.

**Reply by Danny-dK** [reply]
Yes. So first the workflow is to try and delete a vault entry without doing ichmod on the home zone. Depending on ownership and the one that created the vault entry, that could already be feasible. If you find that this does not work, than do the ichmod on the home zone (without -r flag). And then redo the workflow on only the vault entry. Note that vault deletion is only possible for 'rodsadmins' in the system and that this might only be relevant in SURF context.
